clouddriver-saga/src/main/kotlin/com/netflix/spinnaker/clouddriver/saga/events.kt

```diff
@@ -0,0 +1,186 @@
+}
+
+/**
+ * Emitted whenever a [Saga] is saved.
+ *
+ * This event does not attempt to find a difference in state, trading off persistence verbosity for a little bit
+ * of a simpler implementation.
+ *
+ * @param saga The [Saga]'s newly saved state
+ */
+@JsonTypeName("sagaSaved")
+class SagaSaved(
+  val saga: Saga
+) : SagaEvent(saga.name, saga.id)
+
+/**
+ * Emitted whenever an internal error has occurred while applying a [Saga].
+ *
+ * @param reason A human-readable cause for the error
+ * @param error The Exception (if any) that caused the error condition
+ * @param retryable Flags whether or not this error is recoverable
+ * @param data Additional data that can help with diagnostics of the error
+ */
+@JsonTypeName("sagaInternalErrorOccurred")
+class SagaInternalErrorOccurred(
+  sagaName: String,
+  sagaId: String,
+  val reason: String,
+  val error: Exception? = null,
+  val retryable: Boolean = true,
+  val data: Map<String, String> = mapOf()
+) : SagaEvent(sagaName, sagaId)
+
+/**
+ * Emitted whenever an error has occurred within a [SagaAction] while applying a [Saga].
+ *
+ * @param actionName The Java simpleName of the handler
+ * @param error The Exception that caused the error condition
+ * @param retryable Flags whether or not this error is recoverable
+ */
+@JsonTypeName("sagaActionErrorOccurred")
+class SagaActionErrorOccurred(
+  sagaName: String,
+  sagaId: String,
+  val actionName: String,
+  val error: Exception,
+  val retryable: Boolean
+) : SagaEvent(sagaName, sagaId)
+
+/**
+ * Informational log that can be added to a [Saga] for end-user feedback, as well as operational insight.
+ * This is a direct tie-in for the Kato Task Status concept with some additional bells and whistles.
+ *
+ * @param message A tuple message that allows passing end-user- and operator-focused messages
+ * @param diagnostics Additional metadata that can help provide context to the message
+ */
+@JsonTypeName("sagaLogAppended")
+class SagaLogAppended(
+  sagaName: String,
+  sagaId: String,
+  val message: Message,
+  val diagnostics: Diagnostics? = null
+) : SagaEvent(sagaName, sagaId) {
+
+  /**
+   * @param user An end-user friendly message
+   * @param system An operator friendly message
+   */
+  data class Message(
+    val user: String? = null,
+    val system: String? = null
+  )
+
+  /**
+   * @param error An error, if one exists. This must be a [SpinnakerException] to provide retryable metadata
+   * @param data Additional metadata
+   */
+  data class Diagnostics(
+    val error: SpinnakerException? = null,
+    val data: Map<String, String> = mapOf()
+  )
+}
+
+/**
+ * Emitted when all actions for a [Saga] have been applied.
+ */
+@JsonTypeName("sagaCompleted")
+class SagaCompleted(
+  sagaName: String,
+  sagaId: String,
+  val success: Boolean
+) : SagaEvent(sagaName, sagaId)
+
+/**
+ * Emitted when a [Saga] enters compensation (e.g. a rollback operation).
+ */
+@JsonTypeName("sagaInCompensation")
+class SagaInCompensation(
```

Comment:
nit about language...  why not call this `SagaRollbackStarted` and `SagaRollbackCompleted` (for the subsequent). Just more familiar terms, i think